src/osd/PrimaryLogPG.cc

```diff
@@ -10382,32 +10377,22 @@ void PrimaryLogPG::issue_repop(RepGather *repop, OpContext *ctx)
     }
   }
 
-  for (set<pg_shard_t>::const_iterator i = acting_recovery_backfill.begin();
-       i != acting_recovery_backfill.end();
-       ++i) {
-    pg_shard_t peer(*i);
-    if (peer == pg_whoami) continue;
-    if (async_recovery_targets.count(peer) && peer_missing[peer].is_missing(soid)) {
-      for (auto &&entry: ctx->log) {
-	missing_loc.add_missing(soid, ctx->at_version, eversion_t(), entry.is_delete());
```

Comment:
@neha-ojha 

If we modified a snapshot-object, we should clone it first. Hence there should be two log entries in the log list of opctx, one for the head object(whose object-id is soid), and the other is for the cloned object(whose object-id is **not** soid, it should be coid instead, see __PrimaryLogPG::make_writeable__). So we should update missing_loc by the log entry itself here, just like we update peer_missing.